frontend/__tests__/components/AddBookModal/add-book-modal.test.js

```diff
@@ -0,0 +1,70 @@
+import React from "react";
+import { render, screen, userEvent, waitFor } from "../../../test-utils";
+import Button from "../../../components/Button/Button";
+import Text from "../../../components/Text/Text";
+import Modal from "../../../components/Modal/Modal";
+import AddBookModal from "../../../components/AddBookModal/AddBookModal";
+
+jest.mock("../../../components/Modal/Modal");
+describe("Add Book Modal", () => {
+  let onCloseMock;
+  let onSubmitMock;
+  beforeEach(() => {
+    onCloseMock = jest.fn();
+    onSubmitMock = jest.fn();
+    Modal.mockImplementation(
+      ({
+        title,
+        children,
+        onClose,
+        onSubmit,
+        cancelText = "Cancel",
+        submitText = "Submit",
+      }) => (
+        <>
+          <Text>{title}</Text>
+          {children}
+          <Button onClick={onClose}>{cancelText}</Button>
+          <Button onClick={onSubmit}>{submitText}</Button>
+        </>
+      )
+    );
+    render(<AddBookModal onSubmit={onSubmitMock} onClose={onCloseMock} />);
```

Comment:
Please elaborate on why this approach was taken

Reply:
I opted to mock the Modal component by rendering it as a component with two buttons because the Add Book Modal handles the onClose and onSubmit function differently from the Modal. I decided that the easiest way to test the call to both functions was by both of them being called by their respective button. I would click each button in a different test and I could check and verify that both functions were being called correctly. The mocked Modal component also has text to ensure that the title is rendered correctly.